### PR TITLE
#30 backend

### PR DESCRIPTION
**Pull request by DjKrys**
#30

### CODE REVIEW

File: src/sw/Core/src/MainMenuState.cpp [hunk 1]
```diff
@@ -2,26 +2,40 @@
 #include "MainMenuState.h"
 #include "StateManager.h"
 #include "Utils.h"
-
+#include "Room.h"
 #include <memory>
```
Comment (LigasN) [inline src/sw/Core/src/MainMenuState.cpp]: Enter would be nice here

File: src/sw/Core/src/MainMenuState.cpp [hunk 2]
```diff
@@ -2,26 +2,40 @@
...
+	}
+	else
+	{
+		m_guiBuilder.addTextBox(215, 140, "In " + rooms->front()->getName() + " temperature:", 20u);
+		m_guiBuilder.addTextBox(395, 140, std::to_string(rooms->front()->getTemperature()), 20u);
+		m_guiBuilder.addTextBox(435, 140, "*C", 20u);
 
-	m_guiBuilder.addTextBox("SecondRoom temperature:", TEXT_COLOR, 210, 170, 20);
-	m_guiBuilder.addTextBox("23*C", TEXT_COLOR, 405, 170, 20);
+		if (rooms->capacity() > 1)
```
Comment (LigasN) [inline src/sw/Core/src/MainMenuState.cpp]: The comment that it is a temporary solution

File: src/sw/Core/src/NewDeviceControllingSignalMenuState.h [hunk 3]
```diff
@@ -2,22 +2,20 @@
 #define INC_NEWDEVICECONTROLLINGSIGNALMENUSTATE_H_
 
 #include "State.h"
+#include "Room.h"
 
 class NewDeviceControllingSignalMenuState: public State
 {
 	public:
-		NewDeviceControllingSignalMenuState(std::shared_ptr<StateManager> stateManager);
+		NewDeviceControllingSignalMenuState(std::shared_ptr<StateManager> stateManager, std::shared_ptr<Room> room);
```
Comment (LigasN) [inline src/sw/Core/src/NewDeviceControllingSignalMenuState.h]: To new line formatting

File: src/sw/Core/src/Room.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+void Room::setName(std::string name)
+{
+	m_name = std::move(name);
+}
+
+void Room::setTemperature(int temperature)
+{
+	m_temperature_to_change = std::move(temperature);
```
Comment (LigasN) [inline src/sw/Core/src/Room.cpp]: with int, the move is not necessary

File: src/sw/Core/src/TemperatureSensor.cpp [hunk 5]
```diff
@@ -5,9 +5,12 @@ TemperatureSensor::TemperatureSensor(EventManager* eventManager, Sensor sensor)
 {
 	m_eventManager = eventManager;
 	m_GPIOx = GPIOC;
-	m_sensor = sensor;
 }
 
+void TemperatureSensor::setSensor(Sensor sensor)
+{
+	m_sensor = sensor;
+}
```
Comment (LigasN) [inline src/sw/Core/src/TemperatureSensor.cpp:13]: new line after "}"

File: src/sw/Core/src/TemperatureSensor.h [hunk 6]
```diff
@@ -8,13 +8,14 @@
 
 class TemperatureSensor
 {
-	private:
+	protected:
```
Comment (LigasN) [inline src/sw/Core/src/TemperatureSensor.h]: why?

**Review by LigasN — CHANGES_REQUESTED** [review]

File: src/sw/Core/src/MainMenuState.cpp (hunk 1 above)
Comment (DjKrys) [reply src/sw/Core/src/MainMenuState.cpp]: fixed

File: src/sw/Core/src/MainMenuState.cpp (hunk 2 above)
Comment (DjKrys) [reply src/sw/Core/src/MainMenuState.cpp]: fixed

File: src/sw/Core/src/NewDeviceControllingSignalMenuState.h (hunk 3 above)
Comment (DjKrys) [reply src/sw/Core/src/NewDeviceControllingSignalMenuState.h]: fixed

File: src/sw/Core/src/Room.cpp (hunk 4 above)
Comment (DjKrys) [reply src/sw/Core/src/Room.cpp]: fixed

File: src/sw/Core/src/TemperatureSensor.cpp (hunk 5 above)
Comment (DjKrys) [reply src/sw/Core/src/TemperatureSensor.cpp:13]: fixed

File: src/sw/Core/src/TemperatureSensor.h (hunk 6 above)
Comment (DjKrys) [reply src/sw/Core/src/TemperatureSensor.h]: change to private, my mistake

**Review by LigasN — APPROVED** [review]

**Pull request merged**